### PR TITLE
Add checkGlobalNameAvailability to Microsoft.DevOpsInfrastructure/pools

### PR DESCRIPTION
**Pull request by jackylui-ms**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Review by bryanrichter — APPROVED** [review]

File: specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure.Management/main.tsp [hunk 1]
```diff
@@ -748,3 +774,30 @@ model ImageVersionProperties {
...
+  @doc("The name already exists.")
+  "AlreadyExists",
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "No ProvisioningState"
+@doc("The CheckNameAvailability operation response.")
+model CheckNameAvailabilityResult {
+  @doc("True if the name is available, otherwise false.")
+  available: boolean;
+
```
Comment (raosuhas) [inline specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure.Management/main.tsp:794]: consider using enums instead : 

[https://github.com/Azure/azure-rest-api-specs/pull/31422/checks?check_run_id=32572988597](https://github.com/Azure/azure-rest-api-specs/pull/31422/checks?check_run_id=32572988597)

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

File: specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure.Management/main.tsp [hunk 2]
```diff
@@ -268,6 +268,9 @@ model SecretsManagementSettings {
   @doc("Where to store certificates on the machine.")
   certificateStoreLocation?: string;
 
+  @doc("Name of the certificate store to use on the machine, currently 'My' and 'Root' are supported.")
+  certificateStoreName?: string;
```
Comment (raosuhas) [inline specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure.Management/main.tsp:272]: use an enum instead

**Comment by razvanbadea-msft** [comment]
Removing the approval label, you need to follow the breaking changes process from https://aka.ms/brch

**Comment by jackylui-ms** [comment]
> Choose a PR Template description and add a description to this pr to have the Purpose of PR and due diligence sections added

When I click on the Control Plane link, nothing happens. Can you please advise?

**Comment by jackylui-ms** [comment]
> Removing the approval label, you need to follow the breaking changes process from https://aka.ms/brch

I removed the breaking change.

**Comment by razvanbadea-msft** [comment]
@jackylui-ms the breaking change needs to be approved by the Breaking Change Board not by you. there is a form in this link https://aka.ms/brch that you can use

File: specification/devopsinfrastructure/resource-manager/readme.md [hunk 3]
```diff
@@ -91,7 +91,6 @@ This is not used by Autorest itself.
 
 ``` yaml $(swagger-to-sdk)
 swagger-to-sdk:
-  - repo: azure-sdk-for-python
```
Comment (msyyc) [inline specification/devopsinfrastructure/resource-manager/readme.md:94]: context: Python SDK could be generated from Typespec directly so there is no need to keep generation configuration for swagger.

**Review by elizatargithub7 — APPROVED** [review]

**Pull request merged**